### ISSUE TITLE
JSON deserialization error when Content-Type is set but there is no body

### ISSUE THREAD

**Issue by gshackles** [openrasta/openrasta-core#184]
This seems to be a change in behavior between 2.5 and the 2.6 preview...haven't isolated it to a unit test just yet but it's reproducible in our API by making any GET request which has no request body, but sets a `Content-Type: application/json` header. During the processing of the request, this exception gets thrown when `DataContractJsonSerializer.ReadObject()` attempts to read from an empty stream:

```
The codec failed to process the request entity. See the exception below.
System.Runtime.Serialization.SerializationException: Expecting element 'root' from namespace ''.. Encountered 'None'  with name '', namespace ''. 
   at System.Runtime.Serialization.Json.DataContractJsonSerializer.InternalReadObject(XmlReaderDelegator xmlReader, Boolean verifyObjectName)
   at System.Runtime.Serialization.XmlObjectSerializer.ReadObjectHandleExceptions(XmlReaderDelegator reader, Boolean verifyObjectName, DataContractResolver dataContractResolver)
   at System.Runtime.Serialization.Json.DataContractJsonSerializer.ReadObject(XmlDictionaryReader reader)
   at OpenRasta.Codecs.JsonDataContractCodec.ReadFrom(IHttpEntity request, IType destinationType, String paramName)
   at OpenRasta.OperationModel.Hydrators.RequestEntityReaderHydrator.<>c__DisplayClass15_0.<GetReader>b__0(IHttpEntity obj, IType type, String name)
   at OpenRasta.OperationModel.Hydrators.RequestEntityReaderHydrator.<TryReadPayloadAsObject>d__20.MoveNext()
```

While tinkering with it, I was able to work around the issue by [short circuiting inside of the JSON codec like this](https://github.com/openrasta/openrasta-core/compare/master...ololabs:get-with-content-type) but it seems like something that might have changed upstream from that code and might be better solved there versus in individual codecs (also can't say offhand whether this causes other issues or weird behavior as well...) 

I haven't tracked down the upstream behavior change between 2.5 and 2.6 yet, so definitely open to ideas and thoughts on how best address this.

**Comment by gshackles** [comment]
Another possible solution is #186

**Comment by serialseb** [comment]
i had a brain fart. If you skip the request codec when there's no content-length, then you preclude the processing of requests that are transfer-encoded, so it breaks the whole world. I just reverted that commit.

**Comment by serialseb** [comment]
I thought we had done some work on this already. https://github.com/openrasta/openrasta-core/pull/171

**Comment by serialseb** [comment]
Should be fixed in 6074a8d2cd12da95a7b5856e243c90a89f554aa1
